src/XMakeBuildEngine/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -516,7 +516,10 @@ private int LaunchNode(string msbuildLocation, string commandLineArgs)
                 // from `dotnet build3`, using `dotnet.deps.json` and `dotnet.runtimeconfig.json`
                 var runtimeConfigFile = depsFile.Replace(".deps.json", ".runtimeconfig.json");
 
-                dotnetConfigArguments = $"exec --depsfile {depsFile} --runtimeconfig {runtimeConfigFile}";
+                if (File.Exists(depsFile) && File.Exists(runtimeConfigFile))
```

Comment:
Do want want each of these checks separate?  What if the .deps.json file exists, but the runtimeConfig doesn't?